platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationContextImpl.java

```diff
@@ -196,6 +196,47 @@ public boolean cleanDirectory(Path path) {
     return true;
   }
 
+  /**
+   * Temporary version of cleanDirectory() that doesn't delete any directories but only files found.
+   * This is to work around a directory handle issue on Windows which leaves the directory in a
+   * weird state and prevents us from recreating it.
+   *
+   * <p>Cleans a directory without deleting it.
+   *
+   * @param directory directory to clean
+   * @throws IOException in case cleaning is unsuccessful
+   */
+  private static void cleanDirectory(final File directory) throws IOException {
+    final File[] files = directory.listFiles();
+
+    if (files == null) { // null if security restricted
+      throw new IOException("failed to list contents of " + directory);
+    }
+    IOException ioe = null;
+
+    for (final File file : files) {
+      try {
+        if (file.isDirectory()) {
+          ImportMigrationContextImpl.cleanDirectory(file);
+        } else {
+          final boolean exists = file.exists();
+
+          if (!file.delete()) {
+            if (!exists) {
+              throw new FileNotFoundException("file does not exist: " + file);
+            }
+            throw new IOException("unable to delete file: " + file);
+          }
+        }
+      } catch (IOException e) {
+        ioe = e;
```

Comment:
❓ We're ok with only reporting the last error that occurred? What if there are multiple failures and the one that really matters to figure out what the problem is was the first one? Should we keep track of the list of files that caused problem and put that list in the `IOException` being thrown out?

Reply:
Again, almost identical copy of Apache's cleanDirectory(). At this point, I don't think it makes much difference if you report the first or the last. If this was meant to stay here for a longer period, I would probably record in the migration report all errors and not just one.